### ISSUE TITLE
[visual-recognition] updateLocalModel feature request

### ISSUE THREAD

**Issue by bourdakos1** [watson-developer-cloud/swift-sdk#847]
A common pattern when using the new Core ML feature is to periodically check for and download any updates to the specified model. As a developer I would like to show some sort of UI to indicate that the model is being updated. However currently, there is no way to check if the SDK is actually going to update the model before deciding to show the UI.

If you show an indicator immediately before calling `updateLocalModel` and hide it on success or failure the indicator will show for a second while the SDK checks for available updates.

I propose adding an optional closure to the method signature that is called when the model is actually going to update.

```swift
visualRecognition.updateLocalModel(classifierID: modelID, updatesAvailable: {
    // Show model update indicator.
}, failure: { (error: Error) in
    // Hide model update indicator.
    // Handle error.
}, success: {
    // Hide model update indicator.
})
```

I think this might also make it more clear to developers that the function checks for updates before updating. A common question I get is "Is there a function to check if a model needs to be updated before calling `updateLocalModel`?"

**Comment by bourdakos1** [comment]
Although, it might just make more sense to pull out the model check into it's own function 🤷‍♂️
```swift
visualRecognition.checkLocalModelStatus(classifierID: modelID) { (hasUpdates, error) in
    guard let hasUpdates = hasUpdates else {
        // Handle error.
        return
    }
    if hasUpdates {
        // Show model update indicator.
        visualRecognition.updateLocalModel(classifierID: modelID) { (result, error) in
            // Hide model update indicator.
            guard let result = result else {
                // Handle error.
                return
            }
        }
    }
}
```

**Comment by devinaconley** [comment]
@mkistler @mediumTaj - can we talk about this ticket for the next major version?

@bourdakos1 and I spoke on this, and we are thinking the simplest option would be to separate `checkLocalModelForUpdates` and `updateLocalModel` into separate functions. It resolves the control issue and also helps to make the update process clearer to the developer

**Comment by mediumTaj** [comment]
@mkistler suggests we have a `needsUpdate` property of `localModel` would be the cleanest implementation of this. Alternatively this can be checked in `getLocalModel`.
@devinaconley said we should not check for update in `getLocalModel` since we do not want to introduce a network call to this offline method. He suggests we may be able to do the check in `getClassifier` but this is a generated method.

It's problematic - when a classifier is updated, the `retrained` and `updated` field will show a newer date than the local model but until the model status is `ready` only the old Core ML model will be available.

@AnthonyOliveri

**Comment by mkistler** [comment]
I think I would be okay with something like:
```
public func getLocalModelStatus(classifierID: String, completionHandler: ((LocalModelStatus, Error) -> Void))
```

With
```
public struct LocalModelStatus {
    public var needsUpdate: boolean
}
```

This would allow us to return additional info if needed by just expanding the `LocalModelStatus` struct.

**Comment by mediumTaj** [comment]
@devinaconley @bourdakos1 Are you guys OK with this approach?

**Comment by devinaconley** [comment]
Functionally sounds good

Do you think `checkLocalModelStatus` is clear that it's a comparison between local and cloud?

**Comment by AnthonyOliveri** [comment]
@mkistler and I determined that this should not cause any breaking changes, so it will not be included in the Swift 1.0 release. However, this has been sitting in our backlog for awhile, so I plan to implement this not long after 1.0 is released.

**Comment by repjarms** [comment]
@bourdakos1 I'm taking over a lot of the responsibilities on the Swift SDK, can you catch me up and let me know if this is still an open issue?

I'm not sure if @AnthonyOliveri got to this or not.
